docs/scripting-ref.rst

```diff
@@ -391,6 +393,124 @@ Compare:
         -- ...
     end
 
+See also: :ref:`splash-runjs`, :ref:`splash-jsfunc`, :ref:`splash-autoload`,
+:ref:`wait-for-resume`.
+
+.. _wait-for-resume:
+
+splash:wait_for_resume
+----------------------
+
+Run asynchronous JavaScript code in page context. The Lua script will
+yield until the JavaScript code tells it to resume.
+
+**Signature:** ``msg, error = splash:wait_for_resume(snippet, timeout)``
+
+**Parameters:**
+
+* snippet - a string with a JavaScript source code to execute. This code
+  must include a function called ``main``. The first argument to ``main``
+  is an object that has the properties ``resume`` and ``error``. ``resume``
+  is a function which can be used to resume Lua execution. It takes an optional
+  string argument which will be returned to Lua in the ``ok`` return value.
+  ``error`` is a function which can be called with a required string value
+  that is returned in the ``error`` return value.
+* timeout - a float which determines (in seconds) how long to allow JavaScript
+  to execute before cancelling it and returning control to Lua. The timeout
+  must be strictly greater than zero and less than or equal to 60. The timeout
+  defaults to 5 seconds.
+
+**Returns:** ``msg, error`` pair. When the execution is successful
+``msg`` is either nil or contains a stringified JavaScript value. In case of
```

Comment:
Why is the value stringified? Other functions try to return value converted to Lua objects when possible.